python&numpy&pandas&数据可视化matplotlib作业

Jingruizhang1 opened the pull request:
Homework for python&numpy&pandas is presented

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 105
- Score: 97.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 10
- Score: 83.00%
---

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 105
- Score: 97.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 10
- Score: 83.00%
---

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 10
- Score: 83.00%
---

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

iphysresearch commented:
@Jingruizhang1 
很棒！赶快把Python扩展作业也做一下吧！

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

Jingruizhang1 commented:
上传数据可视化matplotlib作业

Jingruizhang1 commented:
提交机器学习作业1

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

iphysresearch commented on 2023/homework/Jingrui_Zhang/homework-matplotlib.ipynb:
Matplotlib 作业！

iphysresearch commented on 2023/homework/Jingrui_Zhang/homework_credit_scoring.ipynb:
Sklearn建模作业！

iphysresearch commented:
@Jingruizhang1 
要不把Python的扩展作业5个题也做一下吧哈？

Jingruizhang1 commented:
python扩展作业：

434. Number of Segments in a String

```
class Solution(object):
    def countSegments(self, s):
        split = s.split() #分隔为各单词（及其后面可能接的符号）组成的list
        return split.__len__() #计数
```

1869. Longer Contiguous Segments of Ones than Zeros

```
class Solution(object):
    def checkZeroOnes(self, s):
        split0 = s.split('0') #用0做分隔符把字符串分隔，得到纯1的列表
        split1 = s.split('1') #用1做分隔符把字符串分隔，得到纯0的列表
        no_of_1 = max(len(i) for i in split0) #计算1列表里最大的部分的位数
        no_of_0 = max(len(i) for i in split1) #计算0列表里最大的部分的位数
        if no_of_1 > no_of_0: #比较两者大小
            return True #严格大于则返回True
        else:
            return False #否则返回False
```

1784. Check if Binary String Has at Most One Segment of Ones

```
class Solution(object):
    def checkOnesSegment(self, s):
        split0 = s.split('0') #用0做分隔符将字符串分割为数字1的列表
        split0_not_empty = [i for i in split0 if i != ''] #将空字符串剔除
        if split0_not_empty.__len__() > 1: #看列表长度是否比1大
            return False #大于1时返回False
        else:
             return True #否则返回True
```

852. Peak Index in a Mountain Array

```
class Solution(object):
    def peakIndexInMountainArray(self, arr):
        left = 0 #初始化左边边界
        right = len(arr) - 1 #初始化右边边界
        while True: #保证此循环一直发生，直到return值
            middle = (left + right) // 2 #初始化中间值，同时每次循环都重新计算中间值
            if arr[middle] > arr[middle - 1] and arr[middle] > arr[middle + 1]: #此时middle处一定为所需值
                return middle #输出middle值
            elif arr[middle] < arr[middle + 1]: #当不满足条件时，判断中间值右边大小，以此判断所需值在左边还是右边
                left = middle + 1 #此时中间值在右边，重新设定左边边界
            else:
                right = middle - 1 #此时中间值在左边，重新设定右边边界
```

162. Find Peak Element

```
class Solution(object):
    def findPeakElement(self, nums):
        left = 0 #初始化左边边界
        right = len(nums) - 1 #初始化右边边界
        if len(nums) == 1: #设定例外：只有一个值
            return 0 #此时返回0
        elif len(nums) == 2: #设定例外：只有两个值
            if nums[0] > nums[1]: #若左大于右
                return 0 #则返回0
            else: #若右大于左（题目中无左等于右的例外）
                return 1 #则返回1
        elif nums[0] > nums[1]: #设定例外/简单情形：最左边的值大于相邻的值
            return 0 #此时返回0
        elif nums[len(nums) - 1] > nums[len(nums) - 2]: #设定例外/简单情形：最右边的值大于相邻的值
            return len(nums) - 1 #此时返回最右边
        else:
            while True: #保证此循环一直发生，直到return值
                middle = (left + right) // 2 #初始化中间值，同时每次循环都重新计算中间值
                if nums[middle] > nums[middle - 1] and nums[middle] > nums[middle + 1]: #此时middle处一定为所需值
                    return middle #输出middle值
                elif nums[middle] < nums[middle + 1]: #当不满足条件时，判断中间值右边大小，以此判断所需值在左边还是右边
                    left = middle + 1 #此时中间值在右边，重新设定左边边界
                else:
                    right = middle - 1 #此时中间值在左边，重新设定右边边界
```

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

iphysresearch commented:
Good Job!!!   @Jingruizhang1
---
Python homework:
- Total questions: 108
- Correct answers: 108
- Score: 100.00%
---
Numpy homework:
- Total questions: 10
- Correct answers: 10
- Score: 100.00%
---
Pandas homework:
- Total questions: 12
- Correct answers: 12
- Score: 100.00%
---

Jingruizhang1 commented:
老师，python扩展作业已提交，机器学习作业2也已提交，谢谢老师的课程，收获很多，老师辛苦了！ @iphysresearch

iphysresearch commented on 2023/homework/Jingrui_Zhang/homework_credit_scoring_finetune_ensemble.ipynb:
Sklearn 模型调优作业！

iphysresearch reviewed (approved):
@Jingruizhang1 
Python基础+拓展作业，Matplotlib数据可视化作业，Sklearn建模和调优作业！
